src/generators/api-links/utils/extractExports.mjs

```diff
@@ -0,0 +1,265 @@
+'use strict';
+
+import { visit } from 'estree-util-visit';
+import { CONSTRUCTOR_EXPRESSION } from '../constants.mjs';
+
+/**
+ * @see https://github.com/estree/estree/blob/master/es5.md#assignmentexpression
+ *
+ * @param {import('acorn').ExpressionStatement} node
+ * @param {string} basename
+ * @param {Record<string, number>} nameToLineNumberMap
+ * @returns {import('../types').ProgramExports | undefined}
+ */
+function handleExpression(node, basename, nameToLineNumberMap) {
+  const { expression } = node;
+
+  if (expression.type !== 'AssignmentExpression') {
+    return;
+  }
+
+  // `a=b`, lhs=`a` and rhs=`b`
+  let { left: lhs, right: rhs, loc } = expression;
+
+  if (lhs.type !== 'MemberExpression') {
+    return undefined;
+  }
+
+  if (lhs.object.type === 'MemberExpression') {
+    lhs = lhs.object;
+  }
+
+  /**
+   * @type {import('../types').ProgramExports}
+   */
+  const exports = {
+    ctors: [],
+    identifiers: [],
+    indirects: {},
+  };
+
+  if (lhs.object.name === 'exports') {
```

Comment:
This is such a big function. I'd rather have a dedicated file for each one of these "cases" or dedicated functions on a separate file, just to reduce the chunk of this function. Pretty much just thinking on reusability.

Then you could have a map:

```js
import identifierHandler ...

const typeHandlers = {
  'Identifier: identifierHandler,
  ...,
};
```

Reply:
This applies for all these large function bodies here. You can create a sub directory. I think it becomes easier to read, easier to maintain. And if you have different functions for handling identifiers or expressions, etc. Simply make multiple exports for a file that handles everything related to a given type.

Reply:
Note that this applies to all the helper functions you've made on all these different files. Let's decrease the size of these functions.